flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveParallelismInference.java

```diff
@@ -63,10 +63,10 @@ int limit(Long limit) {
             return parallelism;
         }
 
-        if (limit != null) {
-            parallelism = Math.min(parallelism, (int) (limit / 1000));
+        if (!infer || limit == null) {
```

Comment:
seems the logic of `!infer` has been covered by the above code
`
if (!infer) {
   return parallelism;
}
`

Reply:
I think we only add  ``` if (!infer && limit == null) { return parallelism; } ```  is better,
then the `limit` still work;

Reply:
@luoyuxia Thank you for your review.
> I mean If set table.exec.hive.infer-source-parallelism = false, it will always return the table.exec.resource.default-parallelism configured by user.
> And if it's -1 , it will fallback to the parallelism of StreamExecutionEnvironment.

if `set table.exec.hive.infer-source-parallelism = false` and `table.exec.resource.default-parallelism` is not set by user it is default value -1, current logic of method `int limit(Long limit) ` return 1, and will not fallback to the parallelism of StreamExecutionEnvironment.

Reply:
The problem occurs in 1.13.1 version. The master branch has already resolved the problem.

Reply:
sorry, this problem has been fixed in master branch. I will close this pr.
https://github.com/apache/flink/blob/2d2d92e9812851091d7cee9c9c1764a0f7b4fdc8/flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveParallelismInference.java#L81